core/unit_test/headers_self_contained/CMakeLists.txt

```diff
@@ -1,19 +1,29 @@
 # Create tests that contain each header separately. We do not  run these tests
 # but we just try to compile them.
+IF ((NOT DEFINED Kokkos_DEFAULT_DEVICE_MEMORY_SPACE OR KOKKOS_DEFAULT_DEVICE_MEMORY_SPACE STREQUAL "") AND
+   (NOT DEFINED Kokkos_DEFAULT_HOST_MEMORY_SPACE OR KOKKOS_DEFAULT_HOST_MEMORY_SPACE STREQUAL "") AND
+   (NOT DEFINED Kokkos_DEFAULT_DEVICE_EXECUTION_SPACE OR KOKKOS_DEFAULT_DEVICE_EXECUTION_SPACE STREQUAL "") AND
+   (NOT DEFINED Kokkos_DEFAULT_HOST_PARALLEL_EXECUTION_SPACE OR KOKKOS_DEFAULT_HOST_PARALLEL_EXECUTION_SPACE STREQUAL ""))
```

Comment:
It was a workaround for when these tests failed after overriding the default...I have since made ither changes that may have resolved this. I will remove and take it from there...